### PR TITLE
Drone factory for malf AI

### PR DESCRIPTION
**Pull request by Perakp**
Gives malf AI the power to build a drone factory for 40 points.
The factory comes with enough materials to build four syndrones. The drones start with a toolbox and instructions to maintain the factory by inserting metal and glass in it.
The circuitboard to construct the factory is not researchable.
A factory that builds regular drones is also available for admins to play with.

Currently the sprite for the machine is a slight edit on the mech fabricator, a little bit darker with a red light. If a spriter wants to do a proper sprite for this, it'd need to have open/closed/active states and animations for inserting metal and glass.

### CODE REVIEW

**Comment by Jalleo** [comment]
:+1: Almost but depending on the health of the syndrones this might need buffing or specific checking for size of alive pop not on the malf side when spawning this factory.

**Comment by JJRcop** [comment]
I don't like that it's a child of the syndrone, the syndrone's "New()" is made for having an uplink if you haven't noticed.

Why not make them both a superclass called /antag/ or something, keep the welcome_message code and ```icon_state = "drone_synd"
	icon_living = "drone_synd"
	picked = TRUE```, but I don't like the other stuff, like for instance the faction setting. I made syndrones to be "drone traitors", the "syndicate" faction is for that. I don't believe it is for malf. Also the "eat other drones to get stronger" shouldn't be in this either. I intentionally made syndrones OP, because they are admin-spawn only, but since this is going to be in gameplay, keeping that OPness is not okay.

File: code/modules/mob/living/simple_animal/friendly/drone.dm [hunk 1]
```diff
@@ -607,3 +608,21 @@
...
 
+
+// malf drone
+/mob/living/simple_animal/drone/syndrone/malf
+	name = "Drone"
+	laws = \
+	"<span class='danger'>ERROR ER0RR $R0RRO$!R41.%%!!(%$^^__+ @#F0E4'STATION OVERRUN, ASSUME CONTROL TO CONTAIN OUTBREAK#*�&110010</span>\n"+\
+	"0. Keep the drone factory in good operating condition.\n"+\
+	"1. Intefere.\n"+\
+	"2. Kill.\n"+\
```
Comment (GunHog) [inline code/modules/mob/living/simple_animal/friendly/drone.dm]: Malf drones are servants of the AI, like borgs. At the very least, they should not have any potential to be hostile to it. Please update these laws to have the drones assist the AI or at the very least, not interfere with it or its borgs.

**Comment by GunHog** [comment]
You seem to have made the factory the child of a "normal" drone factory. Do you intend to make this an R&D buildable item in the future? Oh, and imagine the drone fighting! Normal drones are able to fight these malf ones! Let the drone oilbaths begin!

**Comment by RemieRichards** [comment]
@JJRcop All drones are capable of cannibalism, so these should be also.

**Comment by JJRcop** [comment]
@RemieRichards Sorry, I mean the whole "your maxhealth is larger than your starting health". Sure, all drones are capable of cannibalism, that isn't a feature specific to syndrones, but getting stronger than you usually are for doing so is something I don't think is a good idea for normal gameplay.

**Comment by Dorsisdwarf** [comment]
Is there some way to make it apparent to normal drones what these hostile drones are? Drone Wars might be fun.

**Comment by Deantwo** [comment]
Did you watch StarGate lately or what? Reminds me of the [Replicators](http://en.wikipedia.org/wiki/Replicator_%28Stargate%29), make the laws more specific about maybe ripping parts of the station apart to get more metal for the factor?

**Comment by Perakp** [comment]
todo:
- look at making malf drones more distinct from syndrones (sprite, health, name) 
- give them a message about the malf AI they are bound to
- fix infinite resources by deconstructing/constructing the machine
- get better sprites

Normal drones can tell malf drones apart. Mostly because the alternative would be hell for admins. Further support for drone wars is not planned for now.
I will only add researchable circuitboards to this PR if asked by a headcoder. It's a case of balance, ability to metagame the roundtype and bringing people back in the round.
And no I haven't seen stargate.

**Comment by Dorsisdwarf** [comment]
The only support for dronewars needed is normal drones spotting malf drones - the one thing, the very one thing, that they are allowed to kill.

**Comment by Stickymayhem** [comment]
Very cool. Any way to ensure the malf isn't gambling on ghosts though?

**Comment by Alek2ander** [comment]
I think you underestimate how robust drones are. This power needs to be waaay more expensive. Borg factory level expensive.

**Comment by AlexCrimson** [comment]
Yeah drones are insanely robust. Even just 4 drones could wreck the station. Its a really cool idea though, id love to atleast give it a shot for a while and see how things go.

**Comment by Stickymayhem** [comment]
Yes this should remove the shunt and cost 100 cpu. It's a no brainer really.

**Comment by Dorsisdwarf** [comment]
This isn't as powerful as the borg factory - you're getting optional attraction from bored ghosts, not forced loyal lethal servants with laser rifles.

I'd say a hefty cost, but 100CPU and the shunt is OTT

**Comment by GunHog** [comment]
Drones and Cyborgs each have their own pros and cons. Done power is in stealth and having hands + tools. They are basically tiny, unstunnable greytiding assistants that can vent crawl. They are strong when they take down critical equipment. Consider the choice the AI has to make:

<b> Borg factory</b>
- Produces hackable cyborgs bound by law to serve the AI.
- Cyborgs can communicate with the AI and be coordinated against the crew.
- Cyborgs can not only fight for the AI, but bolster its defenses in a variety of ways.
- Hacked borgs of several module types (not just Secborg/Engiborg) are effective in combat.
- The power cell on the borgs are weak, so they are constantly in need of recharging.
- Costs all your CPU AND removes your shunting power.
- Requires live humans, and by extension at least one robust borg to safely bring back victims.
- An entire force, no matter how large, can be wiped out in seconds from a robotics console.

<b> Drone factory </b>
- Produces highly versatile drones lawed to destroy and reproduce. (They are basically Replicators)
- They have no power cells and have no concerns about running out of energy.
- They are quite mobile, and are not hindered by bolted airlocks due to ventcrawling.
- They are not at risk of being destroyed all at once as no console controls them.
- They do not require human victims, only metal and glass, which can be stolen easily.
- Their combat effectiveness depends on what they are able to steal as a weapon.
- They are (currently) not bound to the AI in any way, cannot communicate with it.
- Their proposed lawset can make them a potential threat to the AI itself.

**Comment by JJRcop** [comment]
>They are (currently) not bound to the AI in any way, cannot communicate with it.
>(currently)

I intend to keep it this way, drones will never be associated with the AI!

**Comment by Jalleo** [comment]
And that will make it easy for people to metagrief the AI and admins will have hell sorting it out. @hornygranny can confirm that fact. This law flaw is sloppy pre policy as it currently is.

**Comment by AlexCrimson** [comment]
Can we not just give the drones special laws to protect their AI master and obey its commands?

**Comment by mrpain666** [comment]
Just out of curiousity, what laws do syndrones have? What control will the AI have over them if they are essentially being spawned by the AI?

**Comment by hornygranny** [comment]
syndrone laws are basically "ruin everything" you should make a subtype that is slaved to the AI

**Comment by JJRcop** [comment]
@mrpain666 This pr does not introduce syndrones, I strongly dislike the fact that the malf drones are a child of syndrones since I made syndrones to be adminspawn, I **never** intended them to be implemented in regular gameplay in the **any way, shape or form**. @Perakp should instead make them /drone/antag/syndrone, and /drone/antag/malf

**Comment by Alek2ander** [comment]
The armory has a vent. That's all you need to know to see how OP drones can be.

**Comment by Chocobro** [comment]
>syndrones
brilliant :+1:

File: code/modules/mob/living/simple_animal/friendly/drone.dm [hunk 2]
```diff
@@ -607,3 +608,21 @@
 	icon_state = "syndrone_item"
 	drone_type = /mob/living/simple_animal/drone/syndrone
 
+
+// malf drone
+/mob/living/simple_animal/drone/syndrone/malf
```
Comment (JJRcop) [inline code/modules/mob/living/simple_animal/friendly/drone.dm]: Like I said before, I do not like that they are children of syndrones.

Comment (Perakp) [inline code/modules/mob/living/simple_animal/friendly/drone.dm]: Don't worry, I'll update this. Thinking of using the default 30 health and a different sprite. Even debating if they should start with a toolbox or not.

**Comment by RemieRichards** [comment]
Maybe give MalfDrones a spark_system datum and have them constantly spark?
Would also prevent them from making <del>forest</del> plasma fires.

File: code/game/machinery/dronefactory.dm [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+		user << "You insert [transfer_amount] [sname] sheet\s into \the [src]."
+		sleep(10)
+		overlays -= "fab-load-[sname]"
+	else
+		user << "\The [src] cannot hold any more [sname] sheet\s."
+
+/obj/machinery/mecha_part_fabricator/dronefactory/emag_act()
+	drone_type = /obj/item/drone_shell/malf
+	circuit_type = /obj/item/weapon/circuitboard/mechfab/dronefactory/malf
+	visible_message("\icon[src] <b>\The [src]</b> beeps: \"System reset to default factory settings.\"")
```
Comment (RemieRichards) [inline code/game/machinery/dronefactory.dm]: >The default is MalfDrones

Hilarious.

**Comment by mrpain666** [comment]
If I where a malf AI and I where to expend my limited resources on a power such as this one, I would like a little more control over it than "ruin everything", maybe some more "controlled chaos" instead. Maybe give them robotic chat access, a law that says "Obey the AI and serve its interests", but at the expense of more CPU, maybe? Something along those lines.

**Comment by JJRcop** [comment]
I've got no problem with giving malf drones robotic talk access, as long as it doesn't also apply to regular maintenance drones as well!

**Comment by mrpain666** [comment]
@JJRcop agreed

**Comment by Jalleo** [comment]
It was removed from drones before for reasons i cant remember but really it seems like you want to stealth buff standard drones somewhat if you give them AI binary.

**Comment by Cheridan** [comment]
Merge conflicts aside, is this ready?

**Comment by AlexCrimson** [comment]
such a shame too

**Pull request closed without merging**